pkg/api/aim/query/query_test.go

```diff
@@ -32,37 +35,119 @@ func (s *QueryTestSuite) SetupTest() {
+		dialector    string
+		expectedSQL  string
+		expectedVars []interface{}
+	}{
+		{
+			name:         "TestRunNameWithoutFunction",
+			query:        `(run.name == 'run')`,
+			dialector:    postgres.Dialector{}.Name(),
+			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" = $1 AND "runs"."lifecycle_stage" <> $2) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			expectedVars: []interface{}{"run", models.LifecycleStageDeleted},
+		},
+		{
+			name:         "TestRunNameWithContainsFunction",
+			query:        `(run.name.contains('run'))`,
+			dialector:    postgres.Dialector{}.Name(),
+			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" LIKE '%run%' AND "runs"."lifecycle_stage" <> $1) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			expectedVars: []interface{}{models.LifecycleStageDeleted},
+		},
+		{
+			name:         "TestRunNameWithStartWithFunction",
+			query:        `(run.name.startswith('run'))`,
+			dialector:    postgres.Dialector{}.Name(),
+			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" LIKE 'run%' AND "runs"."lifecycle_stage" <> $1) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			expectedVars: []interface{}{models.LifecycleStageDeleted},
+		},
+		{
+			name:         "TestRunNameWithEndWithFunction",
+			query:        `(run.name.endswith('run'))`,
+			dialector:    postgres.Dialector{}.Name(),
+			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" LIKE '%run' AND "runs"."lifecycle_stage" <> $1) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			expectedVars: []interface{}{models.LifecycleStageDeleted},
+		},
+		{
+			name:         "TestRunNameWithMatchFunction",
+			query:        `(run.name.match('run'))`,
+			dialector:    postgres.Dialector{}.Name(),
+			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" ~ 'run' AND "runs"."lifecycle_stage" <> $1) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			expectedVars: []interface{}{models.LifecycleStageDeleted},
+		},
+	}
+
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+			pq := QueryParser{
+				Default: DefaultExpression{
+					Contains:   "run.archived",
+					Expression: "not run.archived",
+				},
+				Tables: map[string]string{
+					"runs":        "runs",
+					"experiments": "Experiment",
+				},
+				Dialector: tt.dialector,
+			}
+			parsedQuery, err := pq.Parse(tt.query)
+			assert.Nil(s.T(), err)
+			result := parsedQuery.Filter(
+				s.db.Session(&gorm.Session{DryRun: true}).Model(models.Run{}),
+			).First(&models.Run{})
+			assert.Nil(s.T(), result.Error)
+			assert.Equal(s.T(), tt.expectedSQL, result.Statement.SQL.String())
+			assert.Equal(s.T(), tt.expectedVars, result.Statement.Vars)
+		})
+	}
+}
+
+func (s *QueryTestSuite) TestSqliteDialector_Ok() {
 	tests := []struct {
 		name         string
 		query        string
+		dialector    string
 		expectedSQL  string
 		expectedVars []interface{}
 	}{
 		{
 			name:         "TestRunNameWithoutFunction",
 			query:        `(run.name == 'run')`,
+			dialector:    postgres.Dialector{}.Name(),
```

Comment:
this should be the SQLite dialector everywhere in this test function

Reply:
done.